File: framework/languages/node/packages/stream-connector/src/Contracts/ZlinkStreamEnums.ts
export enum ZlinkStreamTransport {
  WebSocket = 'webSocket',
  WebSocketSecure = 'webSocketSecure'
}

export { ZlinkStreamCodec } from '@zlink-systems/stream-wire';

export enum ZlinkStreamCompression {
  None = 'none',
  Lz4 = 'lz4'
}

export enum ZlinkStreamDispatchMode {
  Manual = 'manual',
  Immediate = 'immediate'
}

/**
 * Connector diagnostics level (spec 26 §4). `Errors` is the default and
 * preserves the connector's established wire behavior (outbound frames carry
 * the flow pair). `Off` applies spec 27 §4: no flow id is generated or
 * attached to outbound frames (header flag 0x10 stays clear), inbound flow
 * fields are neither validated nor installed into received messages
 * (structural length checks are kept), and no trace-only work runs.
 */
export enum ZlinkStreamDiagnosticsLevel {
  Off = 'off',
  Errors = 'errors',
  Normal = 'normal',
  Detailed = 'detailed'
}

export enum ZlinkStreamMessageKind {
  Send = 1,
  Request = 2,
  Response = 3,
  Error = 4,
  Control = 5
}

export enum ZlinkStreamHeaderFlags {
  None = 0,
  HasRequestSeq = 0x01,
  HasMetadata = 0x02,
  PayloadCompressed = 0x04,
  HasCorrelationId = 0x08,
  HasFlowId = 0x10
}

export type ZlinkFlowOrigin = 'Inbound' | 'Timer' | 'Application' | 'Lifecycle';
export type ZlinkStreamCloseReason =
  | 'ClientClose'
  | 'IdleTimeout'
  | 'HeartbeatTimeout'
  | 'ServerDrain'
  | 'ProtocolError'
  | 'TransportError';

export enum ZlinkStreamErrorCode {
  Disconnected = 'disconnected',
  ConfigurationError = 'configurationError',
  ValidationFailed = 'validationFailed',
  RequestTimeout = 'requestTimeout',
  ConnectTimeout = 'connectTimeout',
  FrameDecodeFailed = 'frameDecodeFailed',
  FrameTooLarge = 'frameTooLarge',
  SendFailed = 'sendFailed',
  CompressionFailed = 'compressionFailed',
  DecompressionFailed = 'decompressionFailed',
  UserCallbackFailed = 'userCallbackFailed',
  ObserverFailed = 'observer-failed',
  ObserverDropped = 'observer-dropped',
  ReceivedMessageDropped = 'received-message-dropped',
  RemoteError = 'remoteError'
}

export enum ZlinkStreamConnectionState {
  Created = 'created',
  Connecting = 'connecting',
  Connected = 'connected',
  Reconnecting = 'reconnecting',
  Disconnected = 'disconnected',
  Closed = 'closed'
}

File: framework/languages/dotnet/src/Systems.Zlink.Stream.Connector/Contracts/ZlinkStreamEnums.cs
namespace Systems.Zlink.Stream.Connector.Contracts;

public enum ZlinkStreamTransport
{
    Tcp,
    Tls,
    WebSocket,
    WebSocketSecure,
}

public enum ZlinkStreamCodec : byte
{
    Raw = 0,
    Json = 1,
    MessagePack = 2,
    Protobuf = 3,
}

public enum ZlinkStreamCompression
{
    None,
    Lz4,
}

public enum ZlinkStreamDispatchMode
{
    Manual,
    Immediate,
}

/// <summary>
///     Diagnostics level of the connector. Mirrors the server-side message-flow
///     tracing levels: <see cref="Off" /> disables every trace-only operation,
///     including outbound flow_id/flow_origin stamping and inbound flow capture.
/// </summary>
public enum ZlinkStreamDiagnosticsLevel
{
    Off = 0,
    Errors = 1,
    Normal = 2,
    Detailed = 3,
}

public enum ZlinkStreamMessageKind : byte
{
    Send = 1,
    Request = 2,
    Response = 3,
    Error = 4,
    Control = 5,
}

public enum ZlinkStreamErrorCode
{
    Disconnected,
    ConfigurationError,
    ValidationFailed,
    RequestTimeout,
    ConnectTimeout,
    FrameDecodeFailed,
    FrameTooLarge,
    SendFailed,
    CompressionFailed,
    TlsValidationFailed,
    DecompressionFailed,
    UserCallbackFailed,
    RemoteError,
}

public enum ZlinkStreamConnectionState
{
    Created,
    Connecting,
    Connected,
    Reconnecting,
    Disconnected,
    Closed,
}

/// <summary>
///     Origin of the flow a received message belongs to (stream-connector .NET spec §11).
/// </summary>
/// <remarks>
///     The <c>flow_origin</c> wire values are 1..4 while these ordinals are 0..3. The
///     header codec converts between the two explicitly; never cast this enum to an
///     integer to obtain a wire value.
/// </remarks>
public enum ZlinkStreamFlowOrigin
{
    Inbound,
    Timer,
    Application,
    Lifecycle,
}

public enum ZlinkStreamCloseReason
{
    ClientClose = 0,
    IdleTimeout = 1,
    HeartbeatTimeout = 2,
    ServerDrain = 3,
    ProtocolError = 4,
    TransportError = 5,
}
